# <center>Fundamentals of Data Analysis Project - Tips Dataset</center>

<p align="center"> <img width="400" height="400" src="Pictures/tip1.png" alt="Choice Function Description" /> </p>


###  <font color = "red">Table of Contents</font><a id="table"/>

* [Goals and objectives](#goals)
* [About the Tips Dataset](#about-dataset)
* [Dataset review](#dataset-review)
 * [Description](#description)
 * [Regresion](#regresion)
 * [Analyse](#analyse)
* [Summary](#summary)
* [References](#references)

## Goals and objectives<a id="goals"/>

* Research nad investigate tips dataset
* Use Jupyter notebook and Seaborn with descriptive statistics and plots to describe dataset
* Discusse and analyse the relationship between the total bill and tip amount
* Analyse the relationship between the variables within the dataset
* Conclusion and summary about research and findings

[**<<<**](#table)

## About the Tips Dataset<a id="about-dataset"/>

Tips dataset is one of the fundamental portions of data which is widely used in Python and machine learning. The history behind this dataset is that one waiter form one restaurant in the USA recorded information about each tip he received over a period of a few months of his work. The restaurant, located in a suburban shopping mall, was part of a national chain and served a varied menu. In observance of local law, the restaurant offered to seat in a non-smoking section to guests who requested it. Each record includes a day and time, and taken together, they show the waiter’s work schedule.

[**<<<**](#table)

## Dataset review

To begin with, dataset research is necessary to upload tips dataset into the python and allow by using appropriate libraries and tools to analyse all data. There are two ways to start working in python with this dataset:
* dataset can be downloaded locally into the repository (the folder called Dataset was created for this purpose) and then by using read function from pandas library get all data available for further manipulation or analysis
* using Seaborn module which has a built-in set of different datasets where one of it is tips dataset

In [2]:
# Import all necessary libraries
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Using pandas and read_cvs function load dataset from local folder - method 1
dataset = pd.read_csv("Dataset/tips.csv")
dataset

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

In [3]:
# Using seaborn load tips dataset from its built-in datasets
data = sns.load_dataset("tips")
data

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

Looking at the above two methods it can be clearly seen that no matter what method was used to load dataset the outcome results is the same.  

[**<<<**](#table)

### Description

Having dataset loaded is possible now to start using descriptive statistics and plots to describe tips dataset.  
At the beginning is important to understand what exact data are inside.
By looking at outputs in previos section the head and tail of dataset is visible. Tips dataset contains 244 rows and 7 column of data. Columns contain following variables:

- **total_bill 
- **tip
- **sex
- **smoker
- **day
- **time
- **size

There are also separate commands to display shape or a certain portion of data one by one and achieve the same summary results. 

In [8]:
# Display the shape of the dataset
data.shape

(244, 7)

In [4]:
# Display the head of the dataset
data.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

In [9]:
# Display the tail of the dataset
data.tail()

total_bill   tip     sex smoker   day    time  size
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

Now when is clear what type of data are inside it is time to do some basic statistical summary on dataset.
Very useful function to do this is **describe**. 

In [10]:
# Display basic statistical summary of the dataset
data.describe()

total_bill         tip        size
count  244.000000  244.000000  244.000000
mean    19.785943    2.998279    2.569672
std      8.902412    1.383638    0.951100
min      3.070000    1.000000    1.000000
25%     13.347500    2.000000    2.000000
50%     17.795000    2.900000    2.000000
75%     24.127500    3.562500    3.000000
max     50.810000   10.000000    6.000000

Looking at the table above, it can be seen the major statistical data i.e.:

* the mean bill was 19.79 
* the mean tip 3 
* the mean size of the party was 2.56 people 


[**<<<**](#table)

### Regresion

[**<<<**](#table)

### Analyse

[**<<<**](#table)

## Summary<a id="summary"/>

[**<<<**](#table)

## References<a id="references"/>

1. [Kaggle - dataset](https://www.kaggle.com/jsphyg/tipping)
1. [Seaborn - analyse data](https://towardsdatascience.com/analyze-the-data-through-data-visualization-using-seaborn-255e1cd3948e)
1. []()
1. []()

[**<<<**](#table)